src/main/java/org/threadly/concurrent/wrapper/intercepter/ExecutorTaskIntercepter.java

```diff
@@ -10,31 +11,44 @@
 
 /**
  * <p>Class to wrap {@link Executor} pool so that tasks can be intercepted and either wrapped, or 
- * modified, before being submitted to the pool.  This abstract class needs to have 
- * {@link #wrapTask(Runnable)} overridden to provide the task which should be submitted to the 
- * {@link Executor}.  Please see the javadocs of {@link #wrapTask(Runnable)} for more details 
- * about ways a task can be modified or wrapped.</p>
+ * modified, before being submitted to the pool. This class can be passed a lamba in the 
+ * constructor, or {@link #wrapTask(Runnable)} can be overridden to provide the task which should 
+ * be submitted to the {@link Executor}.  Please see the javadocs of {@link #wrapTask(Runnable)} 
+ * for more details about ways a task can be modified or wrapped.</p>
  * 
  * <p>Other variants of task wrappers: {@link SubmitterSchedulerTaskIntercepter}, 
  * {@link SchedulerServiceTaskIntercepter}, {@link PrioritySchedulerTaskIntercepter}.</p>
  * 
  * @author jent - Mike Jensen
  * @since 4.6.0
  */
-public abstract class ExecutorTaskIntercepter implements SubmitterExecutor {
+public class ExecutorTaskIntercepter implements SubmitterExecutor {
   protected final Executor parentExecutor;
+  protected final Function<Runnable, Runnable> taskManipulator;
   
   protected ExecutorTaskIntercepter(Executor parentExecutor) {
```

Comment:
Would you mind adding a javadoc for this constructor saying that you must override the wrapTask function.  I know there was not one previously, but now that there is two constructors I think it would be worth pointing out what you must do.